commons/all/src/main/java/org/infinispan/commons/dataconversion/AbstractTranscoder.java

```diff
@@ -0,0 +1,76 @@
+package org.infinispan.commons.dataconversion;
+
+import static org.infinispan.commons.logging.Log.CONTAINER;
+
+import java.io.UnsupportedEncodingException;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.infinispan.commons.util.Util;
+
+/**
+ * Class to inherit when implementing transcoders, will handle pre and post processing of the content.
+ *
+ * @since 13.0
+ */
+public abstract class AbstractTranscoder implements Transcoder {
+
+   private static final List<MediaTypeCodec> CODECS = new ArrayList<>();
+
+   static {
+      CODECS.add(new JavaObjectCodec());
+      CODECS.add(new UrlFormCodec());
+      CODECS.add(new RFC4648Codec());
+   }
+
+   /**
+    * Decodes content before doing the transcoding.
+    *
+    * @param content the content.
+    * @param contentType the  {@link MediaType} describing the content.
+    * @return an Object with the content decoded or the content itself if no decoding needed.
+    * @throws UnsupportedEncodingException if an invalid encoding or type is provided.
+    */
+   protected Object decodeContent(Object content, MediaType contentType) throws UnsupportedEncodingException {
+      if (content == null) return null;
+
+      if (contentType == null) throw new NullPointerException("contentType cannot be null!");
```

Comment:
can be replaced with `Objects.requireNonNull(contentType, "contentType cannot be null!")`